src/host/services/IMUService.cpp

```diff
@@ -23,41 +23,138 @@
 */
 
 #include <KBoxLogging.h>
+#include "common/config/KBoxConfig.h"
 #include "common/signalk/SKUpdateStatic.h"
 #include "common/signalk/SKUnits.h"
 #include "IMUService.h"
 
 void IMUService::setup() {
+  _offsetRoll  = 0.0;
+  _offsetPitch = 0.0;
+
   DEBUG("Initing BNO055");
   if (!bno055.begin()) {
     DEBUG("Error initializing BNO055");
   }
   else {
-    DEBUG("Success!");
+    // put BNO055 into config mode
+    bno055.setMode(bno055.OPERATION_MODE_CONFIG);
+    delay(50);
+    // Configure axis mapping (see Bosch manual section 3.4)
+    //bno055.write8(BNO055_AXIS_MAP_CONFIG_ADDR, REMAP_CONFIG_P2); // P0-P7, Default is P1
+    // set axis remap to default KBox mounting starboard hull
+    bno055.write8(bno055.BNO055_AXIS_MAP_CONFIG_ADDR, 0b00001001 );
+    delay(50);
+    //bno055.write8(BNO055_AXIS_MAP_SIGN_ADDR, REMAP_SIGN_P2); // P0-P7, Default is P1
+    // set sign remap to default KBox mounting on starboard hull
+    bno055.write8(bno055.BNO055_AXIS_MAP_SIGN_ADDR, 0b00000000 );
+    delay(50);
+    bno055.setMode( bno055.OPERATION_MODE_NDOF ); // OPERATION_MODE_NDOF_FMC_OFF
+    delay(1000);
+    if (recallCalibration())
+      DEBUG("Success!");
   }
 }
 
 void IMUService::loop() {
-  bno055.getCalibration(&sysCalib, &gyroCalib, &accelCalib, &magCalib);
+  bno055.getCalibration(&_sysCalib, &_gyroCalib, &_accelCalib, &_magCalib);
 
   eulerAngles = bno055.getVector(Adafruit_BNO055::VECTOR_EULER);
 
-  //DEBUG("Calib Sys: %i Accel: %i Gyro: %i Mag: %i", sysCalib, accelCalib, gyroCalib, magCalib);
+  //DEBUG("Calib Sys: %i Accel: %i Gyro: %i Mag: %i", _sysCalib, _accelCalib, _gyroCalib, _magCalib);
   //DEBUG("Attitude roll: %f pitch: %f  Mag heading: %f", eulerAngles.z(), eulerAngles.y(), eulerAngles.x());
 
   SKUpdateStatic<2> update;
   // Note: We could calculate yaw as the difference between the Magnetic
   // Heading and the Course Over Ground Magnetic.
 
-  /* if orientation == MOUNTED_ON_PORT_HULL */
-  double roll, pitch, heading;
-  roll = eulerAngles.z();
-  pitch = eulerAngles.y();
-  heading = fmod(eulerAngles.x() + 270, 360);
+  /* if orientation == MOUNTED_ON_STARBOARD_HULL */
+  _roll = eulerAngles.z() + _offsetRoll;
+  _pitch = eulerAngles.y() + _offsetPitch;
+  _heading = fmod(eulerAngles.x() + 270, 360);
+  //DEBUG("Attitude heel: %.3f pitch: %.3f  Mag heading: %.3f", _roll, _pitch, _heading);
+
+  if (_magCalib >= cfHdgMinCal) {
+    update.setNavigationHeadingMagnetic(SKDegToRad(_heading));
+    _skHub.publish(update);
+  }
 
-  if (sysCalib == 3) {
-    update.setNavigationAttitude(SKTypeAttitude(/* roll */ SKDegToRad(roll), /* pitch */ SKDegToRad(pitch), /* yaw */ 0));
-    update.setNavigationHeadingMagnetic(SKDegToRad(heading));
+	if (_accelCalib >= cfHeelPitchMinCal && _gyroCalib >= cfHeelPitchMinCal) {
```

Comment:
indent problem here.

Reply:
Will take care of all ident things, caused to open the code in Netbeans before configuring :-(